DrawerKit/DrawerKit/Public API/Structs/DrawerBackgroundConfiguration.swift

```diff
@@ -0,0 +1,39 @@
+import UIKit
+
+/// Configuration options for the drawer's background.
+public struct DrawerBackgroundConfiguration {
+    /// BackgroundView builder.
+    internal let make: () -> UIView
+
+    /// BackgroundView handler.
+    internal let handle: (UIView, HandleContext) -> Void
+
+    public init<BackgroundView>(make: @escaping () -> BackgroundView,
+                                handle: @escaping (BackgroundView, HandleContext) -> Void)
+        where BackgroundView: UIView {
+        self.make = make
+        self.handle = { handle($0 as! BackgroundView, $1) }
+    }
+
+    // MARK: - HandleContext
+
+    public struct HandleContext {
+        public let currentY: CGFloat
+        public let containerHeight: CGFloat
+    }
+}
+
+extension DrawerBackgroundConfiguration {
+    /// Dark background
+    public static let dark: DrawerBackgroundConfiguration = .init(
+        make: { () -> UIView in
+            let backgroundView = UIView()
+            backgroundView.backgroundColor = UIColor.black.withAlphaComponent(0.8)
+            backgroundView.alpha = 0
+            return backgroundView
+        },
+        handle: {
```

Comment:
I'm not sure it will be clear what this `handle` means. I'm thinking about the grey horizontal indicator, but it's not that I suppose. Maybe `transitionContext` as it is something that is going to be called continuously with new values during transition?

Reply:
> that is going to be called continuously with new values during transition?

Yes, this is a continuous UI handler.

How about renaming to `configure` ? (since it is frequently used term for updating UI)

Reply:
hm, don't know, `configure` still feels like something that would be expected to be called once before anything else happens. Maybe the most "wordy" way is the best, like `didUpdateDrawerFrame` or something like that?

Reply:
It may just be a matter of taste, but `DrawerBackgroundConfiguration.didUpdateDrawerFrame(backgroundView:context:)` doesn't sound good to me... 🙄 (drawer is not a subject of this sentence) 

If we re-think in protocol delegation naming convention, `DrawerBackgroundDelegate.drawer(_:didUpdateFrame:backgroundView:context:)` will make sense though.

Anyway, I want to call for another vote for this naming.

Reply:
I meant this name to be a name of the closure instead of handle, so it would be:

```swift
DrawerBackgroundConfiguration(
make: { UIView() }
didUpdateDrawerFrame: { centerX, containerHeight in 
 ...
}
)
```

but yeah, it's partially a matter of taste, but as well hw to make it more clear from consumer POV. If we struggle to find something better than `handle` I'd be fine with it.